notebooks/createOmeroFigures.ipynb

```diff
@@ -0,0 +1,487 @@
+   "outputs": [],
+   "source": [
+    "curr_x = 0\n",
+    "curr_y = 0\n",
+    "panels_json = []\n",
+    "column_count = 2\n",
+    "offset = 10\n",
+    "\n",
+    "for c in range(image.getSizeC()):\n",
+    "    curr_x = c * (width + spacing_x) + offset\n",
+    "    for z in range(image.getSizeZ()):\n",
+    "        curr_y = z * (height + spacing_y) + offset\n",
+    "        j = get_panel_json(image_id, curr_x, curr_y, width, height, z, c)\n",
+    "        j['labels'] = get_labels_json(j, c, z)\n",
+    "        panels_json.append(j)\n",
+    "\n",
+    "figure_json['panels'] = panels_json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***Methods to create figure files, and to attach the same to the source images***"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def create_figure_file(figure_json):\n",
+    "\n",
+    "    figure_name = figure_json['figureName']\n",
+    "    if len(figure_json['panels']) == 0:\n",
+    "        raise Exception('No Panels')\n",
+    "    first_img_id = figure_json['panels'][0]['imageId']\n",
+    "\n",
+    "    # we store json in description field...\n",
+    "    description = {}\n",
+    "    description['name'] = figure_name\n",
+    "    description['imageId'] = first_img_id\n",
+    "\n",
+    "    # Try to set Group context to the same as first image\n",
+    "    conn.SERVICE_OPTS.setOmeroGroup('-1')\n",
+    "    i = conn.getObject(\"Image\", first_img_id)\n",
+    "    gid = i.getDetails().getGroup().getId()\n",
+    "    conn.SERVICE_OPTS.setOmeroGroup(gid)\n",
+    "\n",
+    "    json_string = json.dumps(figure_json)\n",
+    "    file_size = len(json_string)\n",
+    "    f = StringIO()\n",
+    "    # f.write(figure_json)\n",
+    "    json.dump(figure_json, f)\n",
+    "\n",
+    "    update = conn.getUpdateService()\n",
+    "    orig_file = create_original_file_from_file_obj(\n",
+    "        f, '', figure_name, file_size, mimetype=\"application/json\")\n",
+    "    fa = omero.model.FileAnnotationI()\n",
+    "    fa.setFile(omero.model.OriginalFileI(orig_file.getId(), False))\n",
+    "    fa.setNs(wrap(JSON_FILEANN_NS))\n",
+    "    desc = json.dumps(description)\n",
+    "    fa.setDescription(wrap(desc))\n",
+    "    fa = update.saveAndReturnObject(fa, conn.SERVICE_OPTS)\n",
+    "    file_id = fa.getId().getValue()\n",
+    "    print \"Figure Created\", file_id\n",
+    "\n",
+    "\n",
+    "def create_original_file_from_file_obj(fo, path, name, file_size, mimetype=None):\n",
+    "    \"\"\"\n",
+    "    This is a copy of the same method from Blitz Gateway, but fixes a bug\n",
+    "    where the conn.SERVICE_OPTS are not passed in the API calls.\n",
+    "    Once this is fixed in OMERO-5 (and we don't need to work with OMERO-4)\n",
+    "    then we can revert to using the BlitzGateway for this method again.\n",
```

Comment:
Can replace this whole doc string with "Creates and saves an OMERO OriginalFile from a file object"